api/client.ts

```diff
@@ -100,15 +107,13 @@ function connect(
   return new Promise((resolve, reject) => {
     // Instantiate the WebSocket, and make the client available when the
     // connection is open.
-    // @ts-ignore The following line is ignored because TS thinks that
-    // WebSocket is not instantiable.
-    const ws = new options.wsclass(url);
+    const ws = new options!.wsclass!(url);
     const handler = createAsyncHandler(callback, resolve, reject);
-    ws.onopen = (evt) => {
-      handler(null, new Client(ws, options));
+    ws.onopen = (_evt) => {
+      handler(null, new Client(ws, options!));
     };
     ws.onclose = (evt) => {
-      handler("cannot connect WebSocket: " + evt.reason, null);
+      handler("cannot connect WebSocket: " + evt.reason, undefined);
```

Comment:
It looks like `value` is optional here so you could probable just remove `undefined` from this callback.